src/pages/WalletPage.vue

```diff
@@ -352,10 +378,88 @@ export default {
       "checkPendingTokens",
       "decodeRequest",
       "generateNewMnemonic",
+      "redeem",
     ]),
     ...mapActions(useCameraStore, ["closeCamera", "showCamera"]),
     ...mapActions(useNWCStore, ["listenToNWCCommands"]),
     ...mapActions(useNPCStore, ["generateNPCConnection", "claimAllTokens"]),
+    knowThisMintOfTokenJson: function (tokenJson) {
+      const mintStore = useMintsStore();
+      // check if we have all mints
+      for (var i = 0; i < tokenJson.token.length; i++) {
+        if (
+          !mintStore.mints.map((m) => m.url).includes(token.getMint(tokenJson))
+        ) {
+          return false;
+        }
+      }
+      return true;
+    },
+    toggleScanner: function () {
+      if (!this.scanningCard) {
+        try {
+          this.ndef = new window.NDEFReader();
+          this.controller = new AbortController();
+          const signal = this.controller.signal;
+          this.ndef
+            .scan({ signal })
+            .then(() => {
+              console.log("> Scan started");
+
+              this.ndef.addEventListener("readingerror", () => {
+                console.error("Argh! Cannot read data from the NFC tag.");
+                notifyError("Argh! Cannot read data from the NFC tag.");
+                this.controller.abort();
+                this.scanningCard = false;
+              });
+
+              this.ndef.addEventListener(
+                "reading",
+                ({ message, serialNumber }) => {
+                  notify(`Serial: ${serialNumber}`);
+                  try {
+                    const decodedTokenLink = new TextDecoder().decode(
+                      message.records[0].data
+                    );
+                    const cashuIndex = decodedTokenLink.indexOf("cashu");
```

Comment:
We should look for "?token=cashu" here, otherwise the search breaks for URLs that contain the word cashu in the domain name (such as wallet.cashu.me)

Reply:
You're absolutely right.